### PR TITLE
Add bond options field.

### PR DESCRIPTION
**Pull request by leongold**
##### SUMMARY
Don't restrict users to specific bond options based on bond mode; allow users to specify both the desired mode and options as free text.

##### ISSUE TYPE
 - Feature Pull Request

### CODE REVIEW

File: lib/ansible/modules/cloud/ovirt/ovirt_host_networks.py [hunk 1]
```diff
@@ -46,7 +46,7 @@
         description:
             - "Dictionary describing network bond:"
             - "C(name) - Bond name."
-            - "C(mode) - Bonding mode."
+            - "C(options) - Bonding options."
```
Comment (machacekondra) [inline lib/ansible/modules/cloud/ovirt/ovirt_host_networks.py]: This breaks backward compatibility.

Comment (leongold) [reply lib/ansible/modules/cloud/ovirt/ovirt_host_networks.py]: What about keeping mode, but also adding options? i.e.:

C(mode) - bonding mode
C(options) - bonding options

Comment (machacekondra) [reply lib/ansible/modules/cloud/ovirt/ovirt_host_networks.py]: Yep, we need to keep mode and add options.

File: lib/ansible/modules/cloud/ovirt/ovirt_host_networks.py [hunk 2]
```diff
@@ -1,7 +1,7 @@
 #!/usr/bin/python
```
Comment (dankenigsberg) [inline lib/ansible/modules/cloud/ovirt/ovirt_host_networks.py:1]: in your commit message, explain why this change is useful (lets users control fine details of Linux bonding)

Comment (machacekondra) [reply lib/ansible/modules/cloud/ovirt/ovirt_host_networks.py:1]: And add some useful examples.

**Review by machacekondra — COMMENTED** [review]
Current implementation isn't idempotent, you need to check also if the options needs update and update it if needed.

File: lib/ansible/modules/cloud/ovirt/ovirt_host_networks.py [hunk 3]
```diff
@@ -160,48 +174,48 @@
...
             'Load balance (balance-xor)',
             None,
             'Dynamic link aggregation (802.3ad)',
-        ][mode_number]
-
-    try:
-        mode_number = int(mode_number)
-        if mode_number >= 1 and mode_number <= 4:
-            if mode_number == 4:
-                options.append(otypes.Option(name='xmit_hash_policy', value='2'))
```
Comment (machacekondra) [inline lib/ansible/modules/cloud/ovirt/ovirt_host_networks.py]: Why did remove you this? I think you have to pass it, because those are deafult options for bond modes 1-4.

Comment (leongold) [reply lib/ansible/modules/cloud/ovirt/ovirt_host_networks.py]: I will check, but I am pretty sure default values will be set even if they're not explicitly specified (being default)

Comment (leongold) [reply lib/ansible/modules/cloud/ovirt/ovirt_host_networks.py]: I was wrong, I'll re-add this

File: lib/ansible/modules/cloud/ovirt/ovirt_host_networks.py [hunk 4]
```diff
@@ -160,48 +175,54 @@
...
 
-def get_mode_type(mode_number):
-    """
-    Adaptive transmit load balancing (balance-tlb): mode=1 miimon=100
-    Dynamic link aggregation (802.3ad): mode=2 miimon=100
-    Load balance (balance-xor): mode=3 miimon=100
-    Active-Backup: mode=4 miimon=100 xmit_hash_policy=2
-    """
+def get_bond_options(mode, usr_opts):
+    MIIMON_100 = dict(miimon='100')
```
Comment (machacekondra) [inline lib/ansible/modules/cloud/ovirt/ovirt_host_networks.py:179]: Any reason to use upper case?

File: lib/ansible/modules/cloud/ovirt/ovirt_host_networks.py [hunk 5]
```diff
@@ -160,48 +175,54 @@
...
+            type=get_type_name(mode_number),
+            value=str(mode_number)
+        )
+    )
 
+    opts_dict = DEFAULT_MODE_OPTS.get(mode, {})
+    opts_dict.update(**usr_opts)
+
+    options.extend(
+        [otypes.Option(name=opt, value=value)
```
Comment (machacekondra) [inline lib/ansible/modules/cloud/ovirt/ovirt_host_networks.py:223]: I am afraid this won't work in compare method bceause you don't pass `type`, but API does fill it, can you please verify idepmtency works correctly?

File: lib/ansible/modules/cloud/ovirt/ovirt_host_networks.py (hunk 4 above)
Comment (leongold) [reply lib/ansible/modules/cloud/ovirt/ovirt_host_networks.py:179]: Just to hint at being a constant.

File: lib/ansible/modules/cloud/ovirt/ovirt_host_networks.py (hunk 5 above)
Comment (leongold) [reply lib/ansible/modules/cloud/ovirt/ovirt_host_networks.py:223]: We've been only passing type for "mode", and I do that still (line#214). I verified this locally.

Comment (machacekondra) [reply lib/ansible/modules/cloud/ovirt/ovirt_host_networks.py:223]: Oh I see :)

**Review by machacekondra — APPROVED** [review]
shipit

**Review by mwperina — APPROVED** [review]
shipit
